packages/gatsby/cache-dir/loader.js

```diff
@@ -146,13 +151,25 @@ export class BaseLoader {
     // }
     this.pageDb = new Map()
     this.inFlightDb = new Map()
+
+    this.staticQueryDb = new Map()
+
+    this.inFlightPromisesByUrlDb = new Map()
+
     this.pageDataDb = new Map()
     this.prefetchTriggered = new Set()
     this.prefetchCompleted = new Set()
     this.loadComponent = loadComponent
     setMatchPaths(matchPaths)
   }
 
+  memoizedGet(url) {
+    if (!this.inFlightPromisesByUrlDb.has(url)) {
+      this.inFlightPromisesByUrlDb.set(url, doFetch(url, `GET`))
```

Comment:
This is never cleaned up (unlike inFlightDb).

Reply:
Possibly a good change would be to unify memoizedGet and inFlightDb, so that they wrap promise they get into cleanup, then one can use both of them the same way.

Reply:
Did some light refactoring and cleaned up a bunch!